packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/AndroidIntentPlugin.java

```diff
@@ -58,13 +59,54 @@ private Bundle convertArguments(Map<String, ?> arguments) {
         bundle.putDouble(key, (Double) value);
       } else if (value instanceof Long) {
         bundle.putLong(key, (Long) value);
+      } else if (value instanceof byte[]) {
+        bundle.putByteArray(key, (byte[]) value);
+      } else if (value instanceof int[]) {
+        bundle.putIntArray(key, (int[]) value);
+      } else if (value instanceof long[]) {
+        bundle.putLongArray(key, (long[]) value);
+      } else if (value instanceof double[]) {
+        bundle.putDoubleArray(key, (double[]) value);
+      } else if (isTypedArrayList(value, Integer.class)) {
+        bundle.putIntegerArrayList(key, (ArrayList<Integer>) value);
+      } else if (isTypedArrayList(value, String.class)) {
+        bundle.putStringArrayList(key, (ArrayList<String>) value);
+      } else if (isStringKeyedMap(value)) {
+        bundle.putBundle(key, convertArguments((Map<String, ?>) value));
       } else {
         throw new UnsupportedOperationException("Unsupported type " + value);
       }
     }
     return bundle;
   }
 
+  private boolean isTypedArrayList(Object value, Class<?> type) {
+    if (!(value instanceof ArrayList)) {
+      return false;
+    }
+    ArrayList list = (ArrayList) value;
+    for (Object o : list) {
+      if (!(o == null || type.isInstance(o))) {
+        return false;
+      }
+    }
+    // We don't want to make claims about the element type of empty lists.
```

Comment:
Hm. Why not? It should be possible to pass an empty list to an argument.

Reply:
We check for list of integers before list of strings and so we would arbitrarily give preference to `ArrayList<Integer>` over `ArrayList<String>` upon receiving an empty list. If we want to support sending typed, empty lists, then we need to encode additional typing information as part of the method call. The values cannot speak for themselves.

Reply:
Even with type erasure, `Bundle` could be using the type information available in the different calls `putStringArrayList` vs `putIntegerArrayList` to prevent us from putting a String ArrayList and then getting an Integer ArrayList (or vice versa) with the same key. But that does not appear to be the case. The code below runs just fine, and writes `Strange...` to the log.
```
Bundle bundle = new Bundle();
bundle.putStringArrayList("myKey", new ArrayList<String>(Arrays.asList("hello", "world")));
ArrayList<Integer> integers = bundle.getIntegerArrayList("myKey");
Object firstInteger = integers.get(0);
if (firstInteger instanceof String) {
  Log.e("myapp", "Strange...");
}
```
So, you're right, we do not need to be careful with empty or `null`-only lists.